lib/super_digit.rb

```diff
@@ -1,15 +1,42 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) if n is magnitude of number
+# Space Complexity - O(n)
 def super_digit(n)
-   
+
+  if n < 10
+    return n
+  end
+
+  sum = 0
+  until n == 0
+    p = n % 10
+    n = n / 10
+    sum += p
+  end
+
+  return super_digit(sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def refined_super_digit(n, k)
```

Comment:
This works, but it's O( (dk)^2), and you can do better with dynamic programming O(d^2 + k^2)